File: lib/src/main/java/com/asyncanimator/launcher/seq/DefaultAnimationSeqHelper.java
package com.asyncanimator.launcher.seq;

/**
 * DefaultAnimationSeqHelper — SeqHelper 的 no-op 基类。
 *
 * <p>对应分析文档 §6.9。feature off 时返回此基类。
 */
public class DefaultAnimationSeqHelper {

    public void addSeqId(android.os.Bundle bundle) {}
    public boolean canFinishRecent() { return true; }
    public boolean canInterceptGesture() { return true; }
    public boolean delayFinishRecents(Runnable r) { if (r != null) r.run(); return false; }
    public void clearFinishRecentsRunnable() {}
    public void resetInterceptState() {}
    public void updateNextFinishSeqIdIfNeed(Object recentsController) {}
    public long getNextFinishSeqId(Object recentsController) { return 0L; }
}

File: lib/src/main/java/com/asyncanimator/launcher/seq/DefaultAnimationSeqHelper.kt
package com.asyncanimator.launcher.seq

/**
 * DefaultAnimationSeqHelper — SeqHelper 的 no-op 基类。
 *
 * 对应分析文档 §6.9。feature off 时返回此基类。
 */
open class DefaultAnimationSeqHelper {

    open fun addSeqId(bundle: android.os.Bundle?) {}
    open val canFinishRecent: Boolean get() = true
    open val canInterceptGesture: Boolean get() = true
    open fun delayFinishRecents(r: Runnable?): Boolean {
        r?.run()
        return false
    }
    open fun clearFinishRecentsRunnable() {}
    open fun resetInterceptState() {}
    open fun updateNextFinishSeqIdIfNeed(recentsController: Any?) {}
    open fun getNextFinishSeqId(recentsController: Any?): Long = 0L
}
